packages/flutter_tools/test/src/test_build_system.dart

```diff
@@ -28,21 +26,21 @@ class TestBuildSystem implements BuildSystem {
       _onRun = null;
 
   final List<BuildResult> _results;
-  final BuildResult _singleResult;
-  final dynamic _exception;
-  final void Function(Target target, Environment environment) _onRun;
+  final BuildResult? _singleResult;
+  final Object? _exception;
+  final void Function(Target target, Environment environment)? _onRun;
   int _nextResult = 0;
 
   @override
   Future<BuildResult> build(Target target, Environment environment, {BuildSystemConfig buildSystemConfig = const BuildSystemConfig()}) async {
     if (_onRun != null) {
-      _onRun(target, environment);
+      _onRun!(target, environment);
     }
```

Comment:
`_onRun?.call(target, environment)`

Reply:
Sure, it's inside the null check so I did the `!`.  I can move it into a local if you're worried about something racing...

Reply:
> In simple cases like here, your best bet is to slap a ! on the use of the field. It seems redundant, but that’s more or less how Dart behaves today.
https://dart.dev/null-safety/understanding-null-safety#working-with-nullable-fields

Reply:
Not racing, but for methods its easier to just conditionally call

Reply:
This would imply replacing the entire null check with the above